### PR TITLE
GEODE-8855: fix hitmiss and stats integration tests

### PR DESCRIPTION
**Pull request by nonbinaryprogrammer**
- organize tests
- remove tests that don't belong in this class
- fix a formatting issue in RedisCommandType
- make the RedisInfoStatsIntegrationTest run against our redis and
native redis
- test all the stats in info

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractHitsMissesIntegrationTest.java [hunk 1]
```diff
@@ -244,28 +392,24 @@ public void testSunionstore() {
     runDiffStoreCommandAndAssertNoStatUpdates("set", (k, v, s) -> jedis.sunionstore(k, v, s));
   }
 
-  // ------------ Hash related commands -----------
-
+  // TODO our implementation updates the stats when it shouldn't
+  @Ignore
```
Comment (jdeppe-pivotal) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractHitsMissesIntegrationTest.java:396]: I've opened GEODE-8857 to fix SMOVE. Please add this Jira to the `@Ignore` here.

File: geode-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/server/InfoNativeRedisAcceptanceTest.java [hunk 2]
```diff
@@ -28,9 +28,4 @@
   public int getPort() {
     return redis.getPort();
   }
-
-  @Override
-  int getExposedPort() {
```
Comment (jdeppe-pivotal) [inline geode-redis/src/acceptanceTest/java/org/apache/geode/redis/internal/executor/server/InfoNativeRedisAcceptanceTest.java:33]: You will probably need to revert this. When run under docker, there are 2 ports in play - one is the external port exposed for outside access and one is the internal port which the external port is proxied to. It's the internal port that this is returning and that's the port that the Info command will report.

**Review by jdeppe-pivotal — CHANGES_REQUESTED** [review]
Looks good other than the small things I commented on.

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractRedisInfoStatsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+    long preSetupCommandsProcessed = Long.valueOf(getInfo(jedis).get(COMMANDS_PROCESSED));
+
+    jedis.set(EXISTING_STRING_KEY, "A_Value");
+    jedis.hset(EXISTING_HASH_KEY, "Field1", "Value1");
+    jedis.sadd(EXISTING_SET_KEY_1, "m1", "m2", "m3");
+    jedis.sadd(EXISTING_SET_KEY_2, "m4", "m5", "m6");
+
+    await().atMost(Duration.ofSeconds(2))
+        .untilAsserted(() -> assertThat(
+            Long.valueOf(getInfo(jedis).get(COMMANDS_PROCESSED)) - numInfoCalled.get() + 1)
```
Comment (jhutchison) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractRedisInfoStatsIntegrationTest.java:88]: these "magic numbers" still seem like they make things more complicated and confusing for anyone coming in in the future.  If it's not possible to incorporate the offset into one fo the delegate methods, can we at least assign them to variables that indicate why we need to add +1 sometimes to these calculations?

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractRedisInfoStatsIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+    jedis2.close();
+    jedis3.close();
+
+    validateConnectedClients(jedis, preTestConnectedClients, 0);
+  }
+
+  // ------------------- Clients Section -------------------------- //
+
+  // TODO fails on our redis but not native redis
```
Comment (ringles) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractRedisInfoStatsIntegrationTest.java:210]: Is this comment accurate? If so, should we have a TODO (and maybe a story)?

File: geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHashCommandsFunctionExecutor.java [hunk 5]
```diff
@@ -102,14 +102,14 @@ public int hstrlen(ByteArrayWrapper key, ByteArrayWrapper field) {
   @Override
   public long hincrby(ByteArrayWrapper key, ByteArrayWrapper field, long increment) {
     return stripedExecute(key,
-        () -> getRedisHash(key, true)
+        () -> getRedisHash(key, false)
```
Comment (ringles) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHashCommandsFunctionExecutor.java:105]: I assume changing "true" to "false" here makes it match the native Redis behavior, right? If so, that should probably be called out in the main PR description.

**Review by ringles — COMMENTED** [review]
Just a couple of clarifications.

File: geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHashCommandsFunctionExecutor.java (hunk 5 above)
Comment (nonbinaryprogrammer) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHashCommandsFunctionExecutor.java:105]: 👍

**Review by jdeppe-pivotal — APPROVED** [review]

**Pull request merged**